src/main/java/upb/dice/twitter/MaxIDChecker.java

```diff
@@ -19,51 +23,59 @@
 public class MaxIDChecker {
     /**
      * This method checks for the maxID for the given keyword, Returns 1 if keyword is new
+     *
      * @param key to check the respective maxID
      * @return the maxID for the given key
      */
     public long keyParse(String key) throws IOException {
+        long result;
         Map<String, String> map = new HashMap<>();
-        try (BufferedReader br = new BufferedReader(new FileReader("Keyword_MaxID1.txt"))) {
-            String line;
-            while ((line = br.readLine()) != null) {
-                String[] splitString = line.split(",");
-                map.put(splitString[0], splitString[1]);
-            }
-            if (map.containsKey(key)) {
-                String id_string = map.get(key);
-                return new BigDecimal(id_string).intValue();
-            }
-            else {
-                return 1L;
-            }
+        File file = new File("Tweets Search Details" + File.separator + "Keyword_MaxID.txt");
+        file.getParentFile().mkdirs();
+        file.createNewFile();
+        BufferedReader br = new BufferedReader(new FileReader("Tweets Search Details" + File.separator + "Keyword_MaxID.txt"));
```

Comment:
1. Please don't use whitespaces within the file or directory names
2. Please don't hard code paths within the methods. It is better to define a `private static final String` attribute for the class in which the path is stored. This eases the maintenance later on.